### ISSUE TITLE
The validity of Markdown in comments is not tested

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#15]
Let's introduce a new `Lint`, which will check the validity of formatting of comments: must be Markdown format. Maybe we can use something like this: https://www.oxygenxml.com/InstData/Editor/SDK/javadoc/ro/sync/exml/workspace/api/markdown/MarkdownValidator.html

**Comment by Marat-Tim** [comment]
Do we have a list of what exactly we want to check? As far as I know, any text is valid markdown. So we want to implement the warnings that are in oxygen editor? If so, maybe you have an example of using this class, because so far I can't use this class properly

I tried this
```java
import ro.sync.exml.workspace.api.markdown.MarkdownValidator;
import ro.sync.exml.workspace.api.markdown.MarkdownValidatorFactory;

import java.io.StringReader;

public class Main {
    public static void main(String[] args) {
        MarkdownValidator markdownValidator = MarkdownValidatorFactory.createSchematronValidator("");
        var validate = markdownValidator.validate(null, new StringReader("# Hello world"));
        System.out.println(validate);
    }
}
```

but so far I'm getting the same error

```
[[SystemID: 
Severity: error
Line: 1
Column: 1
EndLine: -1
EndColumn: -1
Length: -1
Offset: -1
Message: Content is not allowed in prolog.]]
```

On the Internet, people write that such an error usually occurs due to extra characters at the beginning (often the problem was in the BOM), but this is not the case here

Now I'm trying to figure out the code of validator, but the code is quite complicated

**Comment by yegor256** [comment]
@Marat-Tim we should do something similar to what Markdownlint is doing: https://stackoverflow.com/questions/33580413/syntax-code-style-writing-checking-tool-for-markdown-files

**Comment by yegor256** [comment]
@Marat-Tim it's definitely better to reuse an existing linter of Markdown.

**Comment by VasilevNStas** [comment]
@yegor256 @volodya-lombrozo @maxonfjvipon, a note on this one: English grammar/style validation for `.md` files is now wired into the build via **vale** (`.vale.ini` + `vale.yml` workflow, Google/Microsoft/proselint styles, pinned to specific releases per #1180). That covers the "motive" markdown files and any repository `.md`.

The remaining part of this ticket seems to be validating **Markdown inside EO source comments** (not repo `.md` files) — which vale does not parse today. If the intent is only about motive/docs `.md`, this issue is largely resolved; if it's specifically about comments in EO code, I'd suggest narrowing the description accordingly so it's actionable.

**Comment by volodya-lombrozo** [comment]
@yegor256 It's unclear what we are checking here. Is this issue about EO code comments?

**Comment by volodya-lombrozo** [comment]
@VasilevNStas This issue is about checking markdown format in EO code specifically, not about markdown formatting in the project. That said, `vale` is still quite useful for the project

**Comment by VasilevNStas** [comment]
@volodya-lombrozo, got it — thanks for the clarification. So the core of #15 is validating the Markdown **inside EO comments** (the `# comment` text in XMIR), not the repository's `.md` files. The vale integration only covers the latter.

Looking at the existing comment lints (`comments/comment-is-too-wide`, `comment-without-dot`, etc.), they all work off `//o[...]` text in XMIR — so a "markdown-valid comment" lint would follow the same pattern. The practical question is which Markdown rules are meaningful for one-line EO comments (headings/lists don't apply much); a realistic subset might be: no raw HTML, no unclosed code spans, no link syntax errors. I'd suggest we enumerate that subset in a new issue and implement it as a Java/XSL lint reusing an existing Markdown parser (e.g. `commonmark-java` or markdownlint) rather than the Oxygen validator from the original description.

**Comment by VasilevNStas** [comment]
@volodya-lombrozo @yegor256 Just to close the loop: is the narrow subset worth implementing — i.e. a lint that validates EO **comments** (not repo `.md`, which vale already covers) against a small set of Markdown rules (no raw HTML, no unclosed code spans, no broken link syntax)? Happy to implement if you confirm the scope; otherwise we can leave #15 as is.
